jetty-websocket/websocket-javax-server/src/main/java/org/eclipse/jetty/websocket/javax/server/internal/JavaxWebSocketServerContainer.java

```diff
@@ -307,9 +307,22 @@ protected void doStart() throws Exception
     @Override
     public CompletableFuture<Void> shutdown()
     {
-        LifeCycle.stop(sessionTracker);
         CompletableFuture<Void> shutdown = new CompletableFuture<>();
-        shutdown.complete(null);
+        new Thread(() ->
+        {
+            try
+            {
+                LifeCycle.stop(sessionTracker);
+            }
+            catch (Throwable t)
+            {
+                LOG.warn("Error while stopping SessionTracker", t);
+            }
+            finally
+            {
+                shutdown.complete(null);
+            }
+        }).start();
```

Comment:
This thread should be a daemon thread so it doesn't stop the JVM exiting.

I'm also wondering if we should implement `cancel` on the `CompletableFuture` and interrupt the thread if it is called?

We might then need some support in `Graceful.shutdown` or `Server.doStop` to actually do that cancel.... but perhaps that is feature creep for this PR.   So let's just make this thread look OK

Reply:
I think we might have a bigger problem here if `SessionTracker.doStop()` needs to be called by a different thread. We also use the `SessionTracker` on the client which would have the same issue. So we shouldn't store the `SessionTracker` as a bean anymore and its lifecycle needs to be manually managed.

Reply:
Ah yes, `stop()` is synchronized.
I think the SessionTracker should remain as a managed bean so it is stopped normally if the container is stopped, but that shutdown should first remove the tracker as a bean and then stop it in another thread.

Reply:
Wasn't the whole point of this so that we never called the `SessionTracker` stop if the container is stopped as it could block in an `onClosed`? So why would we still have the `SessionTracker` as a managed bean on the container.